packages/fixie-sdk/src/serve-bin.tsx

```diff
@@ -0,0 +1,125 @@
+#! /usr/bin/env node
+
+import './instrument.js';
+import 'dotenv/config';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import { fastify, FastifyInstance, FastifyRequest, FastifyReply } from 'fastify';
+import { Readable } from 'stream';
+import { createRenderContext } from 'ai-jsx';
+import { InvokeAgentRequest } from './types.js';
+import { FixieRequestWrapper } from './request-wrapper.js';
+
+import { createRemoteJWKSet, jwtVerify } from 'jose';
+import path from 'path';
+
+async function serve({
+  packagePath,
+  port,
+  silentStartup,
+}: {
+  packagePath: string;
+  port: number;
+  silentStartup: boolean;
+}): Promise<void> {
+  const Handler = (await import(path.join(process.cwd(), packagePath))).default;
+  const app: FastifyInstance = fastify();
+
+  const fixieApiHost = process.env.FIXIE_API_HOST ?? 'https://app.fixie.ai';
+
+  const getJwks = createRemoteJWKSet(new URL(`${fixieApiHost}/.well-known/jwks.json`));
+
+  app.addHook('onRequest', async (request: FastifyRequest, reply: FastifyReply) => {
+    try {
+      const token = request.headers.authorization?.split(' ')[1];
+      if (typeof token !== 'string') {
+        reply.send(new Error('No authorization header'));
```

Comment:
Shouldn't this be a `4xx` response? Will this code do that?

Reply:
It will now :)